frontend/src/components/VisitRouter/index.js

```diff
@@ -38,13 +40,33 @@ const VisitRouter = observer(() => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const handleSubmitVisit = () => {
+  const handleSubmitVisit = async () => {
     // TODO: logic protecting against concurrent queue entries
     // if existing visit we are coming from QueueTable, so update visit
-    if (existingVisit.id) {
-      participantStore.updateVisit()
-    } else {
-      participantStore.createVisit()
+    try {
+      let isValid = await validateVisitForm(existingVisit)
+      if (existingVisit.id) {
+        !isValid.length
+          ? participantStore.updateVisit()
+          : isValid.map(error =>
+              participantStore.setSnackbarState(
+                `Error location: ${error.name}, Error message: ${error.message}`
+              )
+            )
+      } else {
+        !isValid.length
+          ? participantStore.createVisit()
+          : isValid.map(error =>
+              console.log(
+                participantStore.setSnackbarState(
+                  `Error location: ${error.name}, Error message: ${error.message}`
+                )
+              )
+            )
+      }
+    } catch (err) {
```

Comment:
You can pass this the handleError function. this will I've you a generic "something went wrong" message and then you can put that in the snack bar. 

side note. right now handle error works best for API errors, but we could stand to have a different error handler for non API based errors. that's a TODO though.

Reply:
Do you mean pass in the handleError function to the setSnackbarState()?